### ISSUE TITLE
map_clone doesn't respect MSRV for copied

### ISSUE THREAD

**Issue by Kixunil** [rust-lang/rust-clippy#8276]
### Summary

`.copied()` method was added in 1.36 but if `msrv` is set to less (e.g. 1.29) in clippy.toml the `map_clone` lint fires anyway

### Lint Name

map_clone

### Reproducer

I tried this code:

```rust
[1, 2, 3].iter().map(|x| *x).collect::<Vec<_>>();
```

clippy.toml:

```toml
msrv = "1.29"
```

I saw this happen:

```
warning: you are using an explicit closure for copying elements
 --> src/main.rs:2:5
  |
2 |     [1, 2, 3].iter().map(|x| *x).collect::<Vec<_>>();
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider calling the dedicated `copied` method: `[1, 2, 3].iter().copied()`
  |
  = note: `#[warn(clippy::map_clone)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#map_clone
```

I expected to see this happen:

No lint fires

### Version

```text
rustc 1.58.0 (02072b482 2022-01-11)
binary: rustc
commit-hash: 02072b482a8b5357f7fb5e5637444ae30e423c40
commit-date: 2022-01-11
host: x86_64-unknown-linux-gnu
release: 1.58.0
LLVM version: 13.0.0
```

### Additional Labels

@rustbot label +I-suggestion-causes-error

**Comment by xFrednet** [comment]
FYI for someone looking into this: the msrv for the copied call is `1.36.0`
